0000-net-mac-address.md

```diff
@@ -0,0 +1,72 @@
+- Feature Name: `net_address_mac`
+- Start Date: 2017-07-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a media access control address (MAC) address datatype, `std::net::MacAddr48` to `std::net`.
+
+# Motivation
+[motivation]: #motivation
+
+Currently there is no standard way to communicate physical (or ethernet) addresses when doing network related development in rust, even though `std::net::IpAddrV4` and `std::net::IpAddrV6` exist. The MAC address is, however, a regularly occurring data-structure when doing any type of network related development.
+
+There is also a proliferation of implementations which are not compatible with each other, forcing developers to manually implement the data type again and again, reducing the opportunity for code re-use and convenience. `nom`[1], `libpnet`[2] and `diesel`[3] being a couple of examples.
```

Comment:
Having polled a couple of developers there is one concern about 'non-complex type' crates as an alternative. the `http` crate has immense value, and is packed full of `const`s and `enum`s that would take any developer a lot of time to bootstrap.
Would people opt-in to 'bloating' their `Cargo.toml` files (and imports) just to have a re-usable simple type? `MacAddr48` is just a `[u8; 6]` - as a coder it might just be simpler to do the latter than go to the trouble of finding a crate. To me this is an argument for inclusion in some sort of larger, standardized library, even if it isn't `std` itself?